packages/components/src/components/ResizableBox/ResizableBox.tsx

```diff
@@ -0,0 +1,209 @@
+        }
+
+        #left-div {
+          left: ${`-${reactiveAreaWidth}`};
+        }
+
+        #right-div {
+          right: ${`-${reactiveAreaWidth}`};
+        }
+
+        #bottom-div {
+          bottom: ${`-${reactiveAreaWidth}`};
+        }
+    `
+);
+
+const Child = styled(Resizers)`
+    position: relative;
+    width: 100%;
+    height: 100%;
+    overflow: scroll;
+`
+
+export const ResizableBox = ({
+    children,
+    directions,
+    isLocked = false,
+    width,
+    minWidth = 0,
+    maxWidth,
+    height,
+    minHeight = 0,
+    maxHeight,
+}: ResizableBoxProps) => {
+    const resizableBoxRef = useRef<HTMLDivElement>(null);
+
+    const [newX, setNewX] = useState<number | undefined>(undefined);
+    const [newY, setNewY] = useState<number | undefined>(undefined);
+    const [newWidth, setNewWidth] = useState<number>(width || minWidth || 0);
+    const [newHeight, setNewHeight] = useState<number>(height || minHeight || 0);
+
+    useEffect(() => {
+        // const resizers = Array.from(document.querySelectorAll(
+        //   directions.map(direction => `#${direction}-div`)
+        //   .join(', ')
+        // )) as Array<HTMLDivElement>;
+
+        // Note: adding event listers inside loop doesn't work
+        // resizers.forEach((resizer) => {
+        //   const myCallback = () => handleMouseDown(resizer.id);
+        //   resizer.addEventListener('mousedown', myCallback);
+        // });
+
+        directions.includes('bottom') && document.querySelector('#bottom-div')?.addEventListener('mousedown', () => handleMouseDown('bottom-div'));
+        directions.includes('top') && document.querySelector('#top-div')?.addEventListener('mousedown', () => handleMouseDown('top-div'));
+        directions.includes('left') && document.querySelector('#left-div')?.addEventListener('mousedown', () => handleMouseDown('left-div'));
+        directions.includes('right') && document.querySelector('#right-div')?.addEventListener('mousedown', () => handleMouseDown('right-div'));
+
+        if (resizableBoxRef.current) {
+            const rect = resizableBoxRef.current.getBoundingClientRect(); // TODO sometimes returns incorrect values
+            setNewX(rect.x);
+            setNewY(rect.y);
+            !(width || minWidth) && setNewWidth(rect.width);
+            !(height || minHeight) && setNewHeight(rect.height);
+        }
+    }, [directions, resizableBoxRef.current]);
+
+    const resize = (resizer: string, e: MouseEvent) => {
+        const mouseX = e.clientX;
+        const mouseY = e.clientY;
+
+        const difX = mouseX - (newX ?? 0) - newWidth;
+        const difY = mouseY - (newY ?? 0) - newHeight;
+
+        let result = 0;
+
+        switch (resizer) {
+            // case 'top': // TODO how exactly should it behave?
+            //   break;
+            case 'left': // TODO how exactly should it behave?
+                result = -1*difX;
+                result = result < 0 ? 1 : result;
```

Comment:
There is `1` because I think the minimal width of a box we want to accept when resizing (shrinking the width of a box) and which would make sense is 1px, not zero. Here I calculate the new width of the box. We don't need to calculate height when resizing from left (or also right).

Reply:
Hm, but maybe I should have used `result <= 0 ? 1 : result;` to make sure the result will be greater than 0.